models/report.js

```diff
@@ -63,6 +70,17 @@ schema.post('save', function() {
   }
 });
 
+schema.methods.flag = function() {
+  this.flagged = true;
+  this.markAsRead();
+};
+
+schema.methods.markAsRead = function() {
+  this.read = true;
+  this.checkedOutBy = null;
+  this.checkedOutAt = null;
```

Comment:
This seems like an issue. What if I click a report and go to detail page. That should mark as read but not erase checkout.